app/src/main/java/de/vectordata/skynet/net/PacketHandler.java

```diff
@@ -104,6 +101,12 @@ void handlePacket(byte id, byte[] payload) {
             if (flags != null)
                 if ((message.messageFlags | flags.value()) != message.messageFlags)
                     throw new IllegalStateException(String.format("Incoming channel message lacks required message flags (got: %s, required at least: %d)", message.messageFlags, flags.value()));
+
+            if (message.isCorrupted) {
+                numCorruptedMessages++;
```

Comment:
As mentioned in #30, it is possible to send corrupted messages intentionally using a modified client.

For chat messages you might be able to use something like `MessageState.Corrupted`. You could reuse the rendering for deleted messages and just change the text to _This message could not be decrypted_.

Corrupted messages with `MessageFlags.Loopback` cannot be caused by a third party. In this case you can increment your counter and show a dialog. Please keep in mind that the client won't be able to decrypt messages from a direct channel if the private keys are corrupted.

Reply:
`MessageState`s only exist for ChatMessages, how should other corrupted channel messages be handled?

Reply:
There are only two types of encrypted channel messages without `MessageFlags.Loopback` which are `P20ChatMessage` and `P21MessageOverride`. Both are connected to a chat message and could use `MessageState.Corrupted`.

Reply:
Alright makes sense; I will implement taht